Add aws-athena-connector

canyousayyes opened the pull request:
Hello,

I have created a connector to query AWS S3 bucket data using AWS Athena.

The README.md contains instructions to setup a sample report and the codes are documented. Any feedbacks are welcome :)

diminishedprime commented on aws-athena-connector/src/glue.js:
```diff
@@ -0,0 +1,75 @@
+      case 'string':
+      case 'timestamp':
+        field = fields.newDimension()
+          .setType(types.TEXT);
```
Could this be `types.YEAR_MONTH_DAY`, or one of the other options instead of text?

canyousayyes replied:
Yes I've updated to use `types.YEAR_MONTH_DAY_HOUR`

diminishedprime commented on aws-athena-connector/README.md:
```diff
@@ -0,0 +1,77 @@
+
+Data Studio will automatically crawls the table schema.
+
+Update the data type of `LogDate` from `TEXT` to `DATE(YYYYMMDD)`, then continue.
```
I put a comment below, but I think this step might be necessary if you use `types.YEAR_MONTH_DAY` for timestamps.

canyousayyes replied:
After the data type handling this step is not necessary anymore.

canyousayyes commented:
Hello @diminishedprime , thanks for the review.

I have updated the data type handling for dates.
Now `DATE` and `TIMESTAMP` in Athena will be converted to `YEAR_MONTH_DAY` and `YEAR_MONTH_DAY_HOUR` in Data Studio.
And the I have added logic to convert correct data types during the `getData()` steps.

Please review if the updated implementation looks good to you :) ?

diminishedprime commented:
This looks great, thanks for the Pull Request!
